Makefile

```diff
@@ -121,9 +127,11 @@ build-images: #toil-docker
 
 #<<<
 build-%:
+	printf "\n\nErrors during build-$*: \n" >> $(ERRORLOG)
 	echo "    started build-$*" >> $(LOGFILE)
+	source setup_hosts.sh; \
 	DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 \
-	docker compose -f lib/candigv2/docker-compose.yml -f lib/$*/docker-compose.yml build $(BUILD_OPTS)
+	docker compose -f lib/candigv2/docker-compose.yml -f lib/$*/docker-compose.yml build $(BUILD_OPTS) 1> >(tee -a >(grep -C 2 "[Ee]rror|[Ww]arning" >> $(ERRORLOG)) /dev/null) 2> >(tee -a $(ERRORLOG))
```

Comment:
I'm noticing a lot of unintended behaviour with tee -- notably, that it messes up coloured output, and also with download progress. Is there an alternative we can look into?

Reply:
Unfortunately tee does change the output format, though not directly - Docker, in detecting that it is piping to tee, will "simplify" its output so it can be written to a file. Otherwise, if color codes and other formatting were kept, logs would look like [this](https://imgur.com/ml48nfL). I don't think an alternative to tee would exactly work since the root problem is that Docker's "regular" output simply doesn't work nicely with files. One possible solution I just saw online is manually filtering out ANSI escape codes using sed, which would make logs more readable, but this would make the logging lines very bulky indeed. I will look into whether they can be put into a function again.